src/core/server/saved_objects/migrations/core/build_index_map.ts

```diff
@@ -28,13 +35,20 @@ export function createIndexMap(
   savedObjectSchemas: SavedObjectsSchemaDefinition,
   indexMap: MappingProperties
 ) {
-  const map: { [index: string]: MappingProperties } = {};
+  const map: IndexMap = {};
   Object.keys(indexMap).forEach(type => {
-    const indexPattern = (savedObjectSchemas[type] || {}).indexPattern || defaultIndex;
+    const schema = savedObjectSchemas[type] || {};
+    const script = schema.convertToAliasScript;
+    const indexPattern = schema.indexPattern || defaultIndex;
     if (!map.hasOwnProperty(indexPattern as string)) {
-      map[indexPattern] = {};
+      map[indexPattern] = { typeMappings: {} };
+    }
+    map[indexPattern].typeMappings[type] = indexMap[type];
+    if (script && map[indexPattern].script) {
+      throw Error(`A script has been defined more than once for index pattern "${indexPattern}"`);
```

Comment:
How about being a little more explicit with the error?

`convertToAliasScript has been defined more than once for index pattern "${indexPattern}"`